ph5/clients/ph5availability.py

```diff
@@ -473,22 +491,48 @@ def get_availability(self, station='*', location='*',
                         ph5_seed_station, ph5_loc, ph5_channel = ret
                         ph5_das = st['das/serial_number_s']
                         ph5_channum = st['channel_number_i']
-                        ph5_sample_rate = self.get_sample_rate(st)
-
-                        times = self.ph5.get_availability(
-                            ph5_das, ph5_sample_rate, ph5_channum,
-                            starttime, endtime)
+                        ph5_start_epoch = st['deploy_time/epoch_l']
+                        ph5_stop_epoch = st['pickup_time/epoch_l']
+                        Das_t = self.ph5.query_das_t(
+                            ph5_das,
+                            chan=ph5_channum,
+                            start_epoch=ph5_start_epoch,
+                            stop_epoch=ph5_stop_epoch)
+                        for das in Das_t:
+                            # Sample rates must first be compared than
+                            ph5_sample_rate = das['sample_rate_i']
+                            times = self.ph5.get_availability(
+                                ph5_das, ph5_sample_rate, ph5_channum,
+                                starttime, endtime)
+                            if das['sample_rate_i'] == st['sample_rate_i']:
+                                samplerate_return = das['sample_rate_i']
+                            if times is None:
+                                continue
```

Comment:
This loop mess up sample rate at the result. It's ok if we don't include sample rate, but when we include sample rate, it gives the wrong result. Please look at station 0407 in array 2 and 3 and compare with das 5553 sample rate 100 and 200.

Reply:
The loop was slightly changed to make sure the correct sample rate is used when querying for cut times. 

``` 
                       for das in Das_t:
                            # Sample rates must first be compared than
                            if das['sample_rate_i'] == st['sample_rate_i']:
                                samplerate_return = das['sample_rate_i']
                                ph5_sample_rate = das['sample_rate_i']
                                times = self.ph5.get_availability(
                                        ph5_das, ph5_sample_rate, ph5_channum,
                                        starttime, endtime)
                            if times is None:
                                continue
```

Reply:
It doesn't change sample rate but change time for sample rate. It might be ok depend on the condition for you to filter later. I don't have time to read carefully the logic later. The original test were correct for sample rate included. It your code pass that test, it should be ok.

Reply:
The original test were correct, it was a good idea to look at the data using HDFview. It allowed me to see my confusion. All of the original test now pass. If you think this is the correct logic we maybe need to add this check into PH5toSXML as well. Currently, PH5toSXML can output the incorrect sample rate.